File: frontend/driver-h5/src/api/task.ts
import { get, post } from './request';

export interface TaskListItem {
  id: number;
  taskNo: string;
  taskName?: string;
  status: number;
  origin?: string;
  destination?: string;
  plannedLoadTime?: string;
  plannedArriveTime?: string;
  actualLoadTime?: string;
  actualArriveTime?: string;
  totalQuantity?: number;
  waybillCount?: number;
  customerName?: string;
  mainDriverName?: string;
  plateNumber?: string;
  carrierType?: number;
  prepaidAmount?: number;
  settledAmount?: number;
  carrierCostAmount?: number;
}

export interface TaskSegment {
  id: number;
  segmentNo: number;
  fromLocation?: string;
  toLocation?: string;
  plannedLoadTime?: string;
  plannedArriveTime?: string;
  actualLoadTime?: string;
  actualArriveTime?: string;
  status: number;
  mileage?: number;
}

export interface TaskWaybillItem {
  id: number;
  waybillNo?: string;
  customerName?: string;
  vehicleBrand?: string;
  vehicleModel?: string;
  dealerName?: string;
  quantity: number;
  status: number;
  loadedAt?: string;
  unloadedAt?: string;
  signedAt?: string;
}

export interface TaskDetail extends TaskListItem {
  segments: TaskSegment[];
  items: TaskWaybillItem[];
  remark?: string;
}

export interface TaskListQuery {
  page?: number;
  pageSize?: number;
  status?: number | number[];
  keyword?: string;
}

export interface PageResult<T> {
  list: T[];
  total: number;
  page: number;
  pageSize: number;
}

export function listMyTasks(params: TaskListQuery) {
  return get<PageResult<TaskListItem>>('/task/my', params as Record<string, unknown>);
}

export function getTaskDetail(id: number) {
  return get<TaskDetail>(`/task/${id}`);
}

export function confirmLoad(id: number, payload?: { actualLoadTime?: string; remark?: string }) {
  return post<void>(`/task/${id}/confirm-load`, payload || {});
}

export function depart(id: number, payload?: { remark?: string }) {
  return post<void>(`/task/${id}/depart`, payload || {});
}

export function confirmArrive(id: number, payload?: { actualArriveTime?: string; remark?: string }) {
  return post<void>(`/task/${id}/confirm-arrive`, payload || {});
}

export function signItem(itemId: number, payload?: { signedAt?: string; remark?: string }) {
  return post<void>(`/task/items/${itemId}/sign`, payload || {});
}

export function revertSignItem(itemId: number, payload: { reason: string }) {
  return post<void>(`/task/items/${itemId}/revert-sign`, payload);
}

File: backend/app/modules/driver/schemas/task.py
"""驾驶员任务相关 Schemas"""

from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel, Field


# ============================================================
# 司机动作请求
# ============================================================

class DriverConfirmLoadRequest(BaseModel):
    """确认装车请求（薄层包装：通过创建装车记录 event_type=1 推进 item.status=1）。"""

    actualLoadTime: Optional[datetime] = Field(
        default=None, description="实际装车时间（不传则取当前服务器时间）"
    )
    location: Optional[str] = Field(default=None, max_length=255, description="装车地点")
    photoUrls: List[str] = Field(default_factory=list, max_length=9, description="装车照片")
    remark: Optional[str] = Field(default=None, max_length=255)


class DriverDepartRequest(BaseModel):
    """确认出发请求（task 2→3）。"""

    actualLoadTime: Optional[datetime] = None
    remark: Optional[str] = Field(default=None, max_length=255)


class DriverConfirmArriveRequest(BaseModel):
    """确认到达请求（创建卸车记录 event_type=2 推进 item.status=2，聚合到 task 3→4）。"""

    actualArriveTime: Optional[datetime] = Field(
        default=None, description="实际到达时间"
    )
    location: Optional[str] = Field(default=None, max_length=255, description="卸车地点")
    photoUrls: List[str] = Field(default_factory=list, max_length=9, description="卸车照片")
    remark: Optional[str] = Field(default=None, max_length=255)


class DriverSignItemRequest(BaseModel):
    """单条挂接行签收（item 2/1/0→3，聚合驱动 task 4→5）。"""

    signedAt: Optional[datetime] = Field(default=None, description="签收时间")
    remark: Optional[str] = Field(default=None, max_length=255)


class DriverRevertSignRequest(BaseModel):
    """撤销签收（item 3→2）；reason 必填。"""

    reason: str = Field(min_length=1, max_length=255, description="撤销原因")


# ============================================================
# 任务返回 Schemas
# ============================================================

class DriverTaskListItem(BaseModel):
    """司机任务列表项（精简字段）"""

    id: int
    taskNo: str
    taskName: Optional[str] = None
    status: int
    origin: Optional[str] = None
    destination: Optional[str] = None
    plannedLoadTime: Optional[datetime] = None
    plannedArriveTime: Optional[datetime] = None
    actualLoadTime: Optional[datetime] = None
    actualArriveTime: Optional[datetime] = None
    totalQuantity: int = 0
    waybillCount: int = 0
    customerName: Optional[str] = None
    mainDriverName: Optional[str] = None
    plateNumber: Optional[str] = None
    carrierType: int = 1
    prepaidAmount: float = 0
    settledAmount: float = 0
    carrierCostAmount: Optional[float] = None


class DriverTaskItem(BaseModel):
    """任务下挂接的运单货物行"""

    id: int
    waybillId: int
    waybillNo: Optional[str] = None
    customerName: Optional[str] = None
    vehicleBrand: Optional[str] = None
    vehicleModel: Optional[str] = None
    dealerName: Optional[str] = None
    quantity: int
    status: int
    loadedAt: Optional[datetime] = None
    unloadedAt: Optional[datetime] = None
    signedAt: Optional[datetime] = None


class DriverTaskSegment(BaseModel):
    """任务运输调令/段（保留对前端的最小展示字段）"""

    id: int
    segmentNo: int
    fromLocation: Optional[str] = None
    toLocation: Optional[str] = None
    plannedLoadTime: Optional[datetime] = None
    plannedArriveTime: Optional[datetime] = None
    actualLoadTime: Optional[datetime] = None
    actualArriveTime: Optional[datetime] = None
    status: int = 0
    mileage: Optional[float] = None


class DriverTaskDetail(DriverTaskListItem):
    """任务详情（含明细行与段）"""

    segments: List[DriverTaskSegment] = Field(default_factory=list)
    items: List[DriverTaskItem] = Field(default_factory=list)
    remark: Optional[str] = None
